clustermesh-apiserver/clustermesh/root.go

```diff
@@ -385,7 +383,7 @@ func startServer(
 		Services:     resources.Services,
 		Endpoints:    resources.Endpoints,
 		Backend:      backend,
-		SharedOnly:   !allServices,
+		SharedOnly:   false,
```

Comment:
`SharedOnly` should be set to `true`. Even better, the parameter could be dropped altogether, as all callers would set it to true now.